### PR TITLE
feat: reduce number of requests

### PR DESCRIPTION
**Pull request by lengthmin**
close: #130

### CODE REVIEW

File: src/helpers/api.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ type RequestConfig = {
   headers?: HeadersInit;
 };
 
+const PER_PAGE = 100;
```
Comment (boojack) [inline src/helpers/api.ts:16]: Maybe the minimum value `30` can get more detail trend for those fresh repo. wdyt

Related api doc:
* https://docs.github.com/en/rest/reference/activity#starring

File: src/helpers/api.ts [hunk 2]
```diff
@@ -90,7 +93,7 @@ namespace api {
     const MAX_REQUEST_AMOUNT = 15;
 
     let pageCount = 1;
-    const regResult = /next.*?page=(\d*).*?last/.exec(headerLink);
+    const regResult = /next.*page=(\d*).*?last/.exec(headerLink);
```
[the remark this answers is not in the document]
Comment (lengthmin) [reply src/helpers/api.ts:96]: so we can change the regex to `/next.*&page=(\d*).*last/`

```diff
- /next.*?page=(\d*).*?last/
+ /next.*&page=(\d*).*last/
```

One of the benefits of this is that greedy mode will increase matching speed(pls see: https://blog.csdn.net/lxcnn/article/details/4756030 (in Chinese, but this is a great article)).

File: src/helpers/api.ts (hunk 1 above)
Comment (lengthmin) [reply src/helpers/api.ts:16]: I just realize that this variable would affect the detail records, but in my test:

ant-design/ant-design has 77k star:
whether we set 100 or 30, it's result is:

```js
[{"date":"2021/2/16","count":0},{"date":"2021/2/20","count":2610},{"date":"2021/4/5","count":5280},{"date":"2021/5/15","count":13290},{"date":"2021/5/16","count":39960},{"date":"2022/2/8","count":77603}]
```

bytebase/star-history:
30:
```js
[{"date":"2015/9/24","count":0},{"date":"2016/8/11","count":120},{"date":"2017/5/7","count":300},{"date":"2018/2/8","count":480},{"date":"2018/9/5","count":660},{"date":"2019/4/4","count":840},{"date":"2019/5/30","count":1020},{"date":"2019/7/16","count":1200},{"date":"2019/10/10","count":1380},{"date":"2020/2/17","count":1560},{"date":"2020/5/19","count":1740},{"date":"2020/9/26","count":1920},{"date":"2021/1/25","count":2100},{"date":"2021/5/2","count":2280},{"date":"2021/10/28","count":2460},{"date":"2022/1/26","count":2640},{"date":"2022/2/8","count":2685}]
```
100
```js
[{"date":"2015/9/24","count":0},{"date":"2016/12/20","count":200},{"date":"2017/5/7","count":300},{"date":"2018/2/27","count":500},{"date":"2018/10/30","count":700},{"date":"2019/5/21","count":900},{"date":"2019/6/14","count":1100},{"date":"2019/7/16","count":1200},{"date":"2019/10/16","count":1400},{"date":"2020/3/9","count":1600},{"date":"2020/7/10","count":1800},{"date":"2020/11/26","count":2000},{"date":"2021/1/25","count":2100},{"date":"2021/5/26","count":2300},{"date":"2021/12/3","count":2500},{"date":"2022/2/8","count":2685}]
```

so a good solution is sperate to 2 different vars. one use to request github, another one use to split arrays.

Comment (lengthmin) [reply src/helpers/api.ts:16]: but it seem the logic should change...

I think 100 or 30 both fine, you decide

Comment (boojack) [reply src/helpers/api.ts:16]: > so a good solution is sperate to 2 different vars. one use to request github, another one use to split arrays.

`30` is fine for this PR. We can do the separation in next time.

**Review by boojack — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**